File: Assets/Code/Scripts/DoorScripts/SimpleDoorScript.js
#pragma strict

class SimpleDoor extends Door {
  
  function SimpleDoor( d : GameObject, i : Transform, o : Transform ) {
    super( d, i, o );
  }
  
  function AnimateIn() {
    Level.currentLevel.MovePlayer( outBlocking.position );
    super();
  }
  
  function AnimateOut() {
    Level.currentLevel.MovePlayer( inBlocking.position );
    super();
  }
}

public var destDoor : GameObject;
public var inBlocking : Transform;
public var outBlocking : Transform;

public var door : SimpleDoor;

function Awake() {
  door = new SimpleDoor( destDoor, inBlocking, outBlocking );
}

function OnMouseOver() {
  if( Input.GetMouseButtonDown( 0 ) ) {
    StartCoroutine( door.Enter() );
  }
}

File: Assets/Code/Scripts/DoorScripts/SimpleDoorScript.cs
using UnityEngine;
using System.Collections;

public class SimpleDoorScript : MonoBehaviour {
  
  public class SimpleDoor : Door {
  
    public SimpleDoor( GameObject d, Transform i, Transform o ) : base( d, i, o ) {}
  
    public override void AnimateIn() {
      Level.currentLevel.MovePlayer( outBlocking.position );
      base.AnimateIn();
    }
  
    public override void AnimateOut() {
      Level.currentLevel.MovePlayer( inBlocking.position );
      base.AnimateOut();
    }
  }

  public GameObject destDoor;
  public Transform inBlocking;
  public Transform outBlocking;

  public SimpleDoor door;

  public void Awake() {
    door = new SimpleDoor( destDoor, inBlocking, outBlocking );
  }

  public void OnMouseOver() {
    if( Input.GetMouseButtonDown(0) ) StartCoroutine( door.Enter() );
  }
}